torch/cuda/graphs.py

```diff
@@ -0,0 +1,89 @@
+import ctypes
+import gc
+import torch
+
+from ._utils import _dummy_type
+
+
+if not hasattr(torch._C, '_CudaStreamBase'):
+    # Define dummy base classes
+    torch._C.__dict__['CUDAGraph'] = _dummy_type('CUDAGraph')
+    torch._C.__dict__['graph_pool_handle'] = _dummy_type('graph_pool_handle')
+
+from torch._C import (
+    CUDAGraph,
+    graph_pool_handle
+)
+
+
+class graph(object):
+    r"""Context-manager that captures CUDA work into a :class:`torch.cuda.CUDAGraph`
+    object for later replay.
+
+    See :ref:`CUDA Graphs <cuda-graph-semantics>` for a general introduction,
+    detailed use, and constraints.
+
+    Arguments:
+        graph (torch.cuda.CUDAGraph): Graph object used for capture.
+        pool: Opaque token (returned by a call to :func:`~torch.cuda.graph_pool_handle()` or :meth:`other_Graph_instance.pool()<torch.cuda.CUDAGraph.pool>`) hinting this graph's capture may share memory from the specified pool.
+        stream (torch.cuda.Stream, optional): If supplied, will be set as the current ambient stream in the context.  If not supplied, ``Graph`` sets its own internal side stream as the ambient stream in the context.
```

Comment:
do you need the word `ambient`? `current stream` is enough. Then, in the `if not supplied` section, 
`context manager will acquire a new stream and set it as the current stream`